tests/test_updater_delegation_graphs.py

```diff
@@ -425,6 +425,90 @@ def test_hash_bins_graph_traversal(
         finally:
             self.teardown_subtest()
 
+    @dataclass
+    class SuccinctRolesTestCase:
+        bit_length: int
+        target_path: str
+        expected_visited_bin: str
+
+    succinct_bins_graph: utils.DataSet = {
+        "case 1": SuccinctRolesTestCase(
+            bit_length=1,
+            target_path="othermissingpath",
+            expected_visited_bin="delegated-0",
+        ),
+        "case 2": SuccinctRolesTestCase(
+            bit_length=1,
+            target_path="missingpath",
+            expected_visited_bin="delegated-1",
+        ),
+        "case 3": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="foo",
+            expected_visited_bin="delegated-0",
+        ),
+        "case 4": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="doo",
+            expected_visited_bin="delegated-1",
+        ),
+        "case 5": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="missingpath",
+            expected_visited_bin="delegated-2",
+        ),
+        "case 6": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="bar",
+            expected_visited_bin="delegated-3",
+        ),
+        "many bins case": SuccinctRolesTestCase(
+            bit_length=8,
+            target_path="bar",
+            expected_visited_bin="delegated-fc",
+        ),
+    }
+
+    @utils.run_sub_tests_with_dataset(succinct_bins_graph)
+    def test_succinct_roles_graph_traversal(
+        self, test_data: SuccinctRolesTestCase
+    ) -> None:
+        # Test traversing the delegation tree when succinct roles is used. For a
+        # successful traversal all top level metadata files plus the expected
+        # bin should exist locally and only one bin must be downloaded.
+
+        try:
+            exp_files = [*TOP_LEVEL_ROLE_NAMES, test_data.expected_visited_bin]
+            exp_calls = [(test_data.expected_visited_bin, 1)]
+
+            self.sim = RepositorySimulator()
+            self.sim.add_succinct_roles_to_delegations(
+                "targets", test_data.bit_length, "delegated"
+            )
+            self.sim.update_snapshot()
+
+            self.setup_subtest()
+
+            updater = self._init_updater()
+            # Call explicitly refresh to simplify the expected_calls list
+            updater.refresh()
+            self.sim.fetch_tracker.metadata.clear()
+            # Check that metadata dir contains only top-level roles
+            self._assert_files_exist(TOP_LEVEL_ROLE_NAMES)
+
+            # Looking for a non-existing targetpath forces updater
+            # to visit a correspondning delegated role
```

Comment:
That's the one with the typo:
```suggestion
            # to visit a corresponding delegated role
```